### PR TITLE
feat: content scripts vue support

### PR DESCRIPTION
**Pull request by wensonsmith**
This PR  referred to #17 ， but base on the latest folder structure .

it also minimize changes and add watch options

### CODE REVIEW

File: src/contentScripts/Content.vue [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<template>
+  <div class="flex justify-center h-20 bg-white">
+    <Logo/>
+    <h2>Content Scripts works</h2>
```
Comment (tmkx) [inline src/contentScripts/Content.vue:4]: singular typo

File: vite.content.config.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { resolve } from 'path'
+import { defineConfig } from 'vite'
+import getPagesViteConfig from './vite.config'
+
+const isDev = process.env.NODE_ENV !== 'production'
+const r = (...args: string[]) => resolve(__dirname, ...args)
```
Comment (tmkx) [inline vite.content.config.ts:6]: These can be imported from `scripts/utils.ts`

File: vite.content.config.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+export default defineConfig(({ command }) => {
+  // @ts-expect-error
```
Comment (tmkx) [inline vite.content.config.ts:9]: seems bad

File: vite.content.config.ts [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  return Object.assign(getPagesViteConfig(command), {
```
Comment (antfu) [inline vite.content.config.ts:10]: Use `mergeConfig` from `vite` instead

File: src/contentScripts/index.ts [hunk 5]
```diff
@@ -1,5 +1,14 @@
...
 import { onMessage } from 'webext-bridge'
+import { createApp } from 'vue'
+import 'vue-global-api'
+import App from './Content.vue'
+
+(() => {
+  const app = document.createElement('div')
+  document.body.appendChild(app)
+  createApp(App).mount(app)
+})()
```
[the remark this answers is not in the document]
Comment (wensonsmith) [reply src/contentScripts/index.ts]: i just copy from #17  😅 , fixed

File: vite.content.config.ts (hunk 3 above)
Comment (wensonsmith) [reply vite.content.config.ts]: Already refactored

File: vite.content.config.ts (hunk 4 above)
Comment (wensonsmith) [reply vite.content.config.ts]: I try mergeConfig , but have problems with build.rollupOptions.input . 

so i extract sharedConfig in vite.config.ts

File: vite.content.config.ts (hunk 2 above)
Comment (wensonsmith) [reply vite.content.config.ts]: Yep , fixed

**Comment by antfu** [comment]
Thanks @wensonsmith @tshemeng! 

@tshemeng I have added you as a co-author of this PR.

**Pull request merged**